### PR TITLE
Migrate detekt-sample-extensions/build.gradle to kotlin dsl

### PR DESCRIPTION
**Pull request by epeee**
Another pr towards #611.

I can work on the remaining two build.gradle files later on (detekt-gradle-plugin and the rootproject one).

### CODE REVIEW

File: detekt-sample-extensions/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+buildscript {
+	repositories {
+		mavenLocal()
+		jcenter()
+	}
+
+	val kotlinVersion by project
```
Comment (arturbosch) [inline detekt-sample-extensions/build.gradle.kts]: buildscript cant pick up the top level var?

**Comment by arturbosch** [comment]
the gradle-plugin gets a rework in #771, so this has to wait for that, but later yes please do it :)

**Comment by Mauin** [comment]
Even with the gradle plugin rework changing the `build.gradle` file is not the worst of the world. Should be fairly easy to rebase, so no reason from my side to not convert the `build.gradle` file to Kotlin as soon as we want 👍

File: detekt-sample-extensions/build.gradle.kts (hunk 1 above)
Comment (epeee) [reply detekt-sample-extensions/build.gradle.kts]: Yep, it can't.

**Pull request merged**